src/function/scalar/list/list_reverse.cpp

```diff
@@ -0,0 +1,124 @@
+#include "duckdb/common/types/data_chunk.hpp"
+#include "duckdb/function/scalar/nested_functions.hpp"
+#include "duckdb/planner/expression/bound_function_expression.hpp"
+#include "duckdb/planner/expression/bound_parameter_expression.hpp"
+#include "duckdb/planner/expression_binder.hpp"
+
+namespace duckdb {
+
+static void ListReverseFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	D_ASSERT(args.ColumnCount() == 1);
+	auto count = args.size();
+
+	Vector &input_list = args.data[0];
+	if (input_list.GetType().id() == LogicalTypeId::SQLNULL) {
+		result.SetVectorType(VectorType::CONSTANT_VECTOR);
+		ConstantVector::SetNull(result, true);
+		return;
+	}
+
+	// get the child vector
+	auto input_list_list_size = ListVector::GetListSize(input_list);
+	auto &input_list_child = ListVector::GetEntry(input_list);
+
+	UnifiedVectorFormat input_list_data;
+	input_list.ToUnifiedFormat(count, input_list_data);
+	auto input_list_entries = (list_entry_t *)input_list_data.data;
```

Comment:
please use `UnifiedVectorFormat::GetData<list_entry_t>(input_list_data);` to get the list entries
We're in the process of banning all c-style casts :)